### PR TITLE
[Notebook] Embed enhancements

### PR DESCRIPTION
**Pull request by jvigliotta**
Closes https://github.com/nasa/openmct/issues/5810 

### Describe your changes:

1. Action button changes to 3-dot menu. 
2. Menu now includes new or updated actions: Quick View, Navigate to Item, Navigate to Item In Time, Remove This Embed. 
3. Selecting these actions should perform the following actions: 
- Quick View: Opens a full screen overlay of the embed. 
- Navigate to Item in Time: Navigates to item in its time frame when captured. 
- Navigate to Item: Navigates to item with the current time settings. 
- Remove This Embed: Removes embed from the notebook entry.
4. A scrolling container has been added to the embed structure when many embeds have been added to an entry. 
5. Multiple CSS/style class additions and changes to enhance visuals. 

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate unit tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Commit messages meet standards?
* [x] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Review by charlesh88 — CHANGES_REQUESTED** [review]
Looking good, there are a few things that need some sanding and shimming:

Screen Shot 2022-10-06 at 4 39 56 PM
- [ ] Remove (trash can) icon should be vertically centered in this row, but isn't. This is because `c-ne__time-and-creator-and-delete` is `display: block`, but should be `display: flex` with `align-items: center`. Child markup _may_ need to be restructured to allow the delete button to right-align. Once moved, we _might_ look at tightening up the vertical margin between that row and the entry input, but probably only by a pixel or two.

Screen Shot 2022-10-06 at 4 41 46 PM

- [ ] Tag menu is getting clipped. This is because `c-ne__time-and-content` is set to `overflow: auto`, but should be `overflow:visible`.

Screen Shot 2022-10-06 at 4 42 57 PM

- [ ] Text descriptions for "Navigate to *" could be better. For 'Navigate to Item', I'd use 'Navigate to the item with the **current** time settings.' For 'Navigate to Item in Time', I'd use 'Navigate to the item **in its time frame when captured**.'

Screen Shot 2022-10-06 at 4 43 17 PM

- [ ] The icon for 'Navigate to Item in Time' is wrong (red arrow) - it should be the same as its object type as seen in 'Navigate to Item' (green arrow). 

Screen Shot 2022-10-06 at 4 46 55 PM


- [ ] We'd discussed changing the inset shadow (orange arrow) and insetting the scroll area  (yellow arrow) with padding so that the scroll bar isn't right on the shadow edge. I'd use the same inset shadow style exactly as applied to the entry input (green arrow). 

Screen Shot 2022-10-06 at 4 48 04 PM

- [x] The 3 dot button should always right-align in the space afforded to the embed. Just add `flex: 1 1 auto` on `.c-ne__embed__link`.

**Comment by rukmini-bose** [comment]
@charlesh88 In regard to the last PR review comments, I have fixed all. Just some comments: 

Regarding inset on the container and scroll area, I was able to use the exact same inset settings that we use in the Notebook input. However, I am not too sure how we should add padding to the scroll bar in order to make the right side of the container look less rectangular, and have more of the rounded corners. 

Screen Shot 2022-10-08 at 5 22 41 PM

In addition, I had to add a new class with some CSS styling to the tags code. Specifically, I had to add `overflow:visible` to this class in order to stop the label from clipping.

**Review by charlesh88 — CHANGES_REQUESTED** [review]
All responded changes look good! Is this PR still intended to be draft? Think that needs to be changed.

> However, I am not too sure how we should add padding to the scroll bar in order to make the right side of the container look less rectangular, and have more of the rounded corners.

- [ ] Checkout this codepen: https://codepen.io/charlesh88/pen/ExLBRME. Note that corner radius and padding is _not_ what we should use in the Notebook, but are just provided to illustrate the solution.

### A few more things that need to be addressed:

- [x] The embed 3 dot button needs some margin, like $interMarginSm: 

Screen Shot 2022-10-20 at 6 55 32 AM

- [ ] `scrollContainer` needs to be changed: CSS class names use the convention of `c-something-something__element` in all lowercase, the single letter prefix (most of the time is 'c' for component) and dashes for spaces.  If this is intended to be a general-purpose class, then it should be moved into `_controls.scss` and shouldn't be in `notebook.scss`. If we want to keep it Notebook-specific, then we should rename as an element of `c-ne`, like `c-ne__embed-scroll-container`.
- [x] For the inset shadow in scrollContainer, just use `nice-input` like this:

Screen Shot 2022-10-20 at 8 11 59 AM

**Comment by charlesh88** [comment]
1. I am still not seeing the scroll area with some margin against the outer drop shadow (as shown in https://codepen.io/charlesh88/pen/ExLBRME), and I don't see changes in the markup for this either:

image

2. Guess I'm not being as clear as I can be about CSS naming conventions: just tacking on a "c_" to the front of `scrollContainer` wasn't what I was after. Names should be _all lowercase_ with dashes, not camelCase. Good that we moved it into `_controls`, that's the right spot.

Let's do a Teams session on this so we can wrap this up.

**Review by charlesh88 — APPROVED** [review]
Great work!  LGTM.

**Comment by shefalijoshi** [comment]
On testing, I clicked on the 3-dot menu and saw this:
image

The right hand side being empty looks a bit odd.

File: src/plugins/notebook/components/NotebookEmbed.vue [hunk 1]
```diff
@@ -88,38 +87,88 @@ export default {
...
+
+            const menuOptions = {
+                menuClass: 'c-ne__embed__actions-menu',
+                placement: this.openmct.menus.menuPlacement.TOP_RIGHT
+            };
+
+            this.openmct.menus.showSuperMenu(x, y, this.menuActions, menuOptions);
+        },
+        addMenuActions() {
+            if (this.embed.snapshot) {
```
Comment (shefalijoshi) [inline src/plugins/notebook/components/NotebookEmbed.vue:114]: If `this.embed.snapshot` is falsy, should `this.menuActions` be set to an empty array?

File: src/plugins/notebook/components/NotebookEmbed.vue [hunk 2]
```diff
@@ -88,38 +87,88 @@ export default {
...
+                };
+
+                this.menuActions.push(removeEmbed);
             }
 
         },
+        async setEmbedObjectPath() {
+            this.objectPath = await this.openmct.objects.getOriginalPath(this.embed.domainObject.identifier);
+
+            if (this.objectPath[this.objectPath.length - 1].type === 'root') {
```
Comment (shefalijoshi) [inline src/plugins/notebook/components/NotebookEmbed.vue:168]: Is the length of `this.objectPath` guaranteed to be at least `1`? If not, this will break.

**Review by shefalijoshi — CHANGES_REQUESTED** [review]
A couple of small changes, but it looks good.

**Comment by rukmini-bose** [comment]
> On testing, I clicked on the 3-dot menu and saw this: image
> 
> The right hand side being empty looks a bit odd.

This is using the super menu component that's used throughout Open MCT in areas such as the create menu and time conductor. These menus keep the description area blank when no option is hovered. I think that the amount of work it'd probably take to create a new component in which the description box is not visible unless hovered, outweighs the way it looks visually. Thoughts?

File: src/plugins/notebook/components/NotebookEmbed.vue (hunk 1 above)
Comment (jvigliotta) [reply src/plugins/notebook/components/NotebookEmbed.vue:114]: I don't think so, this only happens once on mounted, so it should be an empty array already, this menu item only pertains to snapshots as well, so other items are added after.

File: src/plugins/notebook/components/NotebookEmbed.vue (hunk 2 above)
Comment (jvigliotta) [reply src/plugins/notebook/components/NotebookEmbed.vue:168]: not entirely sure to be honest, but to be safe, I put a check in there.

**Review by unlikelyzero — CHANGES_REQUESTED** [review]
We're going to need to add some tests to this one. Shouldn't take more than 3 hours

**Review by shefalijoshi — APPROVED** [review]
Nice!

**Comment by shefalijoshi** [comment]
@jvigliotta @rukmini-bose @unlikelyzero I've approved the changes. Please feel free to merge one the tests are done.

**Comment by unlikelyzero** [comment]
I'm presently writing a test for this. We have no tests in this area so I'm starting from scratch. I've already found one bug that I need to determine if it's on both master and on this branch

**Review by unlikelyzero — APPROVED** [review]

**Pull request merged**